waspc/cli/src/Wasp/Cli/Command.hs

```diff
@@ -1,24 +1,40 @@
+{-# LANGUAGE GADTs #-}
 {-# LANGUAGE GeneralizedNewtypeDeriving #-}
 
 module Wasp.Cli.Command
   ( Command,
     runCommand,
     CommandError (..),
+
+    -- * Requirements
+    require,
+    DbConnectionRequirement (DbConnectionRequirement),
+    WaspRootRequirement (WaspRootRequirement),
```

Comment:
I would prefer to not export the constructors for the requirements, because it would be nice to prevent people from constructing them without going through `require`. I tried using a unidirectional pattern synonym:

```hs
data DbConnectionRequirement = DbConnectionRequirementConstr
pattern DbConnectionRequirement <- DbConnectionRequirementConstr
```

But GHC complains about the pattern being able to fail when you match on it, and `{-# COMPLETE #-}` wasn't helping.

I don't think this is a big deal though, we can just trust people to not misuse these.

Reply:
Got it, makes sense. I remember I used https://stackoverflow.com/questions/8172548/is-it-possible-to-export-constructors-for-pattern-matching-but-not-for-construc for similar problems -> can't remember if I used View Patterns or Pattern Synonyms, maybe the former I think.
Interesting that it complains though hm! I think I got it working ok.
But ok, it is not terrible if we expose constructors, I agree.